src/pocketmine/Player.php

```diff
@@ -1356,16 +1356,12 @@ protected function processMovement($tickDiff){
 
 			$this->move($dx, $dy, $dz);
 
-			$diffX = $this->x - $newPos->x;
-			$diffY = $this->y - $newPos->y;
-			$diffZ = $this->z - $newPos->z;
-
 			$yS = 0.5 + $this->ySize;
-			if($diffY >= -$yS or $diffY <= $yS){
```

Comment:
The original code seems to have logic problems. I think it should be an `and` operator such that it can be expressed as `-$yS <= $diffY <= $yS` mathematically